cmd/util/cmd/checkpoint-trie-stats/cmd.go

```diff
@@ -0,0 +1,110 @@
+package checkpoint_trie_stats
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/onflow/flow-go/ledger/complete/mtrie/node"
+	"github.com/onflow/flow-go/ledger/complete/mtrie/trie"
+	"github.com/onflow/flow-go/ledger/complete/wal"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
+	"github.com/spf13/cobra"
+)
+
+var (
+	flagCheckpoint string
+	flagTrieIndex  int
+)
+
+var Cmd = &cobra.Command{
+	Use:   "checkpoint-trie-stats",
+	Short: "List the trie node count by types in a checkpoint, show total payload size",
+	Run:   run,
+}
+
+func init() {
+
+	Cmd.Flags().StringVar(&flagCheckpoint, "checkpoint", "",
+		"checkpoint file to read")
+	_ = Cmd.MarkFlagRequired("checkpoint")
+	Cmd.Flags().IntVar(&flagTrieIndex, "trie-index", 0, "trie index to read, 0 being the first trie, -1 is the last trie")
+
+}
+
+func run(*cobra.Command, []string) {
+
+	log.Info().Msgf("loading checkpoint %v, reading %v-th trie", flagCheckpoint, flagTrieIndex)
+	res, err := scanCheckpoint(flagCheckpoint, flagTrieIndex, log.Logger)
+	if err != nil {
+		log.Fatal().Err(err).Msg("fail to scan checkpoint")
+	}
+	log.Info().
+		Str("TrieRootHash", res.trieRootHash).
+		Int("InterimNodeCount", res.interimNodeCount).
+		Int("LeafNodeCount", res.leafNodeCount).
+		Int("TotalPayloadSize", res.totalPayloadSize).
+		Msgf("successfully scanned checkpoint %v", flagCheckpoint)
+}
+
+type result struct {
+	trieRootHash     string
+	interimNodeCount int
+	leafNodeCount    int
+	totalPayloadSize int
+}
+
+func readTrie(tries []*trie.MTrie, index int) (*trie.MTrie, error) {
+	if len(tries) == 0 {
+		return nil, errors.New("No tries available")
+	}
+
+	if index < -len(tries) || index >= len(tries) {
+		return nil, fmt.Errorf("index %d out of range", index)
+	}
+
+	if index < 0 {
+		return tries[len(tries)+index], nil
+	}
+
+	return tries[index], nil
+}
+
+func scanCheckpoint(checkpoint string, trieIndex int, log zerolog.Logger) (result, error) {
+	tries, err := wal.LoadCheckpoint(flagCheckpoint, &log)
+	if err != nil {
+		return result{}, fmt.Errorf("error while loading checkpoint: %w", err)
+	}
+
+	log.Info().Msgf("checkpoint loaded, total tries: %v", len(tries))
```

Comment:
```suggestion
	log.Info().
	    Int("total_tries", len(tries)).
	    Msg("checkpoint loaded")
```
I find this easier to search for in the code if I need to.